### PR TITLE
OP-780 - oh core make spring use xxx ACT (rebased)

### PR DESCRIPTION
**Pull request by goto-eof**
See OP-780.
Paired with https://github.com/informatici/openhospital-gui/pull/1334.

### CODE REVIEW

File: sql/step_XX_drop_adm_deleted.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE patient DROP COLUMN pat_deleted;
```
Comment (dbmalkovsky) [inline sql/step_XX_drop_adm_deleted.sql:1]: The file name looks wrong `step_XX....`    and the contents of this file is exactly the same as the next sql file

File: sql/step_XX_drop_pat_deleted.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+ALTER TABLE patient DROP COLUMN pat_deleted;
```
Comment (dbmalkovsky) [inline sql/step_XX_drop_pat_deleted.sql:1]: Again the file name

File: sql/step_XX_drop_adm_deleted.sql (hunk 1 above)
Comment (dbmalkovsky) [inline sql/step_XX_drop_adm_deleted.sql:1]: Note the file name: `step_XX_`  and the contents of this file is the same as the next: `ALTER TABLE patient ...`

File: sql/step_04_all_following_steps.sql [hunk 3]
```diff
@@ -67,5 +67,7 @@ source step_70_increase_examination_note_field.sql;
 source step_71_drop_opd_date_vis_field.sql;
 source step_72_hospital_changes.sql;
 source step_73_ward_visit_duration.sql;
+source step_72_drop_pat_deleted.sql;
+source step_73_drop_adm_deleted.sql;
```
Comment (dbmalkovsky) [inline sql/step_04_all_following_steps.sql:71]: By using step 72 and 73 you are replacing those that already exist there: `step_72_hospital_changes` and `step_73_ward_visit_duration`; please use other numbers.

File: src/main/java/org/isf/patient/service/PatientIoOperationRepositoryImpl.java [hunk 4]
```diff
@@ -113,12 +112,10 @@ public List<Patient> getPatientsByParams(Map<String, Object> params) {
 		Root<Patient> patient = query.from(Patient.class);
 
 		// Only not deleted patient
-		Predicate deletedN = cb.equal(patient.get("deleted"), "N");
-		Predicate deletedNull = cb.isNull(patient.get("deleted"));
-		Predicate notDeleted = cb.or(deletedN, deletedNull);
+		Predicate deletedN = cb.equal(patient.get("active"), Integer.valueOf(1));
```
Comment (mwithi) [inline src/main/java/org/isf/patient/service/PatientIoOperationRepositoryImpl.java:115]: I would rename into `notActive`

File: src/test/java/org/isf/medicalstock/test/Tests.java [hunk 5]
```diff
@@ -245,12 +246,16 @@ public void testIoNewAutomaticDischargingMovementLotQuantityLessMovementQuantity
...
+System.out.println("ZZZ");
+System.out.println("1");
 		Medical medical = foundMovement.getMedical();
+		System.out.println("2");
 		MovementType movementType = foundMovement.getType();
 		Ward ward = foundMovement.getWard();
 		Supplier supplier = foundMovement.getSupplier();
+		System.out.println("3");
 		Lot lot2 = testLot.setup(medical, false); //we are going to create a second lot
+		System.out.println("4");
```
Comment (mwithi) [inline src/test/java/org/isf/medicalstock/test/Tests.java:258]: remove all println...

File: sql/step_04_all_following_steps.sql [hunk 6]
```diff
@@ -69,3 +69,5 @@ source step_72_hospital_changes.sql;
 source step_73_ward_visit_duration.sql;
 source step_74_operation_report_button.sql;
 source step_75_add_prefix_to_table_names.sql;
+source step_76_drop_adm_deleted.sql;
+source step_77_drop_pat_deleted.sql;
```
Comment (mwithi) [inline sql/step_04_all_following_steps.sql:73]: I would combine into one single step, like `step_76_drop_deleted_fields_use_active`

**Review by mwithi — CHANGES_REQUESTED** [review]
LGTM, anyway small changes are requested. Thanks!

**Comment by mwithi** [comment]
it seems there are some conflicts too

File: sql/step_04_all_following_steps.sql [hunk 7]
```diff
@@ -69,3 +69,5 @@ source step_72_hospital_changes.sql;
 source step_73_ward_visit_duration.sql;
 source step_74_operation_report_button.sql;
 source step_75_add_prefix_to_table_names.sql;
+source step_76_drop_adm_deleted.sql;
```
Comment (dbmalkovsky) [inline sql/step_04_all_following_steps.sql:72]: This needs to be renumbered again as there is a recently merged `step_76...`

**Comment by mwithi** [comment]
> it seems there are some conflicts too

ok, just the script name and one import in the PatientExamination class

**Comment by dbmalkovsky** [comment]
The `load_demo_data.sql` script needs to be updated too.

**Comment by dbmalkovsky** [comment]
Why is the `SUP_DELETED` column in the `Supplier` object not included in this PR?   See this issue: https://openhospital.atlassian.net/browse/OP-527.  Doesn't it follow the same pattern as the other two?

**Comment by mwithi** [comment]
> Why is the `SUP_DELETED` column in the `Supplier` object not included in this PR? See this issue: https://openhospital.atlassian.net/browse/OP-527. Doesn't it follow the same pattern as the other two?

yes, good catch! 
actually, the columns drop is a step that could have been done "just after" OP-780 and not "inside" it
the OP-527 is indeed "blocked by" OP-780...

**Comment by goto-eof** [comment]
> Why is the `SUP_DELETED` column in the `Supplier` object not included in this PR? See this issue: https://openhospital.atlassian.net/browse/OP-527. Doesn't it follow the same pattern as the other two?



> > Why is the `SUP_DELETED` column in the `Supplier` object not included in this PR? See this issue: https://openhospital.atlassian.net/browse/OP-527. Doesn't it follow the same pattern as the other two?
> 
> yes, good catch! actually, the columns drop is a step that could have been done "just after" [OP-780](https://openhospital.atlassian.net/browse/OP-780) and not "inside" it the [OP-527](https://openhospital.atlassian.net/browse/OP-527) is indeed "blocked by" [OP-780](https://openhospital.atlassian.net/browse/OP-780)...

done.

File: sql/step_81_drop_adm_deleted.sql [hunk 8]
```diff
@@ -0,0 +1 @@
+ALTER TABLE oh_admission DROP COLUMN pat_deleted;
```
Comment (dbmalkovsky) [inline sql/step_81_drop_adm_deleted.sql:1]: Shouldn't the column that is being dropped be `adm_deleted`?  `pat_deleted` doesn't exist in the table.

**Comment by dbmalkovsky** [comment]
Why isn't removing `SUP_DELETED` from `OH_SUPPLIER` not included in this PR?

**Comment by goto-eof** [comment]
> Why isn't removing `SUP_DELETED` from `OH_SUPPLIER` not included in this PR?

it was a forgetfulness. I added the missing script, fixed the other one and tested them.

**Comment by mwithi** [comment]
> > Why isn't removing `SUP_DELETED` from `OH_SUPPLIER` not included in this PR?
> 
> it was a forgetfulness. I added the missing script, fixed the other one and tested them.

As I was saying above, the deletion should be done in a single step.

**Comment by mwithi** [comment]
Some conflicts to be resolved

File: sql/step_04_all_following_steps.sql [hunk 9]
```diff
@@ -74,6 +74,4 @@ source step_77_laboratory_button_in_admission_patient.sql;
 source step_78_drop_lab_exam_date.sql;
 source step_79_rename_pat_photo_field.sql;
 source step_80_add_ward_opd.sql;
-source step_81_drop_adm_deleted.sql;
-source step_82_drop_pat_deleted.sql;
```
Comment (dbmalkovsky) [inline sql/step_04_all_following_steps.sql]: Step 81 already is being used.

**Review by mwithi — APPROVED** [review]

File: src/main/java/org/isf/admission/service/AdmissionIoOperations.java [hunk 10]
```diff
@@ -264,7 +259,7 @@ public int getNextYProg(String wardId) throws OHServiceException {
 	public static boolean afterJune;
 
 	public static LocalDateTime getNow() {
-		LocalDateTime now = TimeTools.getNow();
+		LocalDateTime now = LocalDateTime.now();
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/admission/service/AdmissionIoOperations.java:262]: Please do not change this; the method `getNow()` is used so nanoseconds are not included in the time and thus can be used successfully across operating systems and architectures without issue.

File: src/main/java/org/isf/admission/service/AdmissionIoOperations.java [hunk 11]
```diff
@@ -232,15 +227,15 @@ public int getNextYProg(String wardId) throws OHServiceException {
...
 			} else {
-				first = now.withMonth(Month.JULY.getValue()).withDayOfMonth(1).with(LocalTime.MIN).truncatedTo(ChronoUnit.SECONDS);
-				last = now.plusYears(1).withMonth(Month.JUNE.getValue()).withDayOfMonth(30).with(LocalTime.MAX).truncatedTo(ChronoUnit.SECONDS);
+				first = now.withMonth(Month.JULY.getValue()).withDayOfMonth(1).with(LocalTime.MIN);
+				last = now.plusYears(1).withMonth(Month.JUNE.getValue()).withDayOfMonth(30).with(LocalTime.MAX);
 			}
 		} else {
-			first = now.with(firstDayOfYear()).with(LocalTime.MIN).truncatedTo(ChronoUnit.SECONDS);
-			last = now.with(lastDayOfYear()).with(LocalTime.MAX).truncatedTo(ChronoUnit.SECONDS);
+			first = now.with(firstDayOfYear()).with(LocalTime.MIN);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/admission/service/AdmissionIoOperations.java:237]: Please do not change this; see note at line 262

File: src/main/java/org/isf/admission/service/AdmissionIoOperations.java (hunk 10 above)
Comment (goto-eof) [reply src/main/java/org/isf/admission/service/AdmissionIoOperations.java]: I made the rebase incorrectly

File: src/main/java/org/isf/admission/service/AdmissionIoOperations.java [hunk 12]
```diff
@@ -232,11 +228,11 @@ public int getNextYProg(String wardId) throws OHServiceException {
 
 		if (wardId.equalsIgnoreCase("M") && GeneralData.MATERNITYRESTARTINJUNE) {
 			if (now.getMonthValue() < Month.JUNE.getValue()) {
-				first = now.minusYears(1).withMonth(Month.JULY.getValue()).withDayOfMonth(1).with(LocalTime.MIN).truncatedTo(ChronoUnit.SECONDS);
-				last = now.withMonth(Month.JUNE.getValue()).withDayOfMonth(30).with(LocalTime.MAX).truncatedTo(ChronoUnit.SECONDS);
+				first = now.minusYears(1).withMonth(Month.JULY.getValue()).withDayOfMonth(1).with(LocalTime.MIN);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/admission/service/AdmissionIoOperations.java:231]: Please put back the `.truncatedTo()` methods in this line and through line 235

File: src/main/java/org/isf/admission/service/AdmissionIoOperations.java [hunk 13]
```diff
@@ -124,9 +123,7 @@ public AdmittedPatient loadAdmittedPatient(final Integer patientId) {
 		if (patient == null) {
 			return null;
 		}
-		if (isLoadPatientProfilePhotoFromDb) {
-			Hibernate.initialize(patient.getPatientProfilePhoto());
-		}
+		Hibernate.initialize(patient.getPatientProfilePhoto());
```
Comment (mwithi) [inline src/main/java/org/isf/admission/service/AdmissionIoOperations.java:126]:
```suggestion
		if (isLoadPatientProfilePhotoFromDb) {
			Hibernate.initialize(patient.getPatientProfilePhoto());
		}
```

File: src/main/java/org/isf/admission/service/AdmissionIoOperations.java [hunk 14]
```diff
@@ -40,7 +40,6 @@
 import org.isf.generaldata.GeneralData;
 import org.isf.patient.model.Patient;
 import org.isf.patient.service.PatientIoOperationRepository;
-import org.isf.patient.service.PatientIoOperations;
 import org.isf.utils.db.TranslateOHServiceException;
```
Comment (mwithi) [inline src/main/java/org/isf/admission/service/AdmissionIoOperations.java:43]:
```suggestion
import org.isf.patient.service.PatientIoOperations;
import org.isf.utils.db.TranslateOHServiceException;
```

File: sql/step_82_drop_deleted.sql [hunk 15]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE oh_admission DROP COLUMN adm_deleted;
+ALTER TABLE oh_patient DROP COLUMN pat_deleted;
+ALTER TABLE oh_supplier DROP COLUMN sup_deleted;
```
Comment (mwithi) [inline sql/step_82_drop_deleted.sql:3]: Sorry, just realized that:

The script, before to deleted those fields, should also "migrate" previous data, by setting `XXX_ACTIVE=0` for all `XXX_DELETED='Y'`.

`load_demo_data.sql` should also be provided.

**Comment by mwithi** [comment]
Merging into a informatici repository

**Pull request merged**